java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightEndpoint.java

```diff
@@ -54,13 +56,27 @@ public FlightEndpoint(Ticket ticket, Location... locations) {
    * Constructs a new endpoint with an expiration time.
    *
    * @param ticket A ticket that describe the key of a data stream.
+   * @param expirationTime (optional) When this endpoint expires.
    * @param locations  The possible locations the stream can be retrieved from.
    */
   public FlightEndpoint(Ticket ticket, Instant expirationTime, Location... locations) {
+    this(ticket, expirationTime, "", locations);
+  }
+
+  /**
+   * Constructs a new endpoint with an expiration time.
+   *
+   * @param ticket A ticket that describe the key of a data stream.
+   * @param expirationTime (optional) When this endpoint expires.
+   * @param appMetadata (optional) Application metadata associated with this endpoint.
+   * @param locations  The possible locations the stream can be retrieved from.
+   */
+  public FlightEndpoint(Ticket ticket, Instant expirationTime, String appMetadata, Location... locations) {
```

Comment:
Should I add a constructor for `FlightEndpoint(Ticket ticket, String appMetadata, Location... locations)`?

Reply:
We can omit that. A builder could be added.

Reply:
Changed to private and added builder.